src/lib.rs

```diff
@@ -19,69 +17,94 @@ static MONGODB: OnceCell<Database> = OnceCell::new();
 
 #[arma]
 fn init() -> Extension {
-    let mut builder = Builder::from_default_env();
+    let mut builder = env_logger::Builder::from_default_env();
     builder.target(Target::Stdout);
     builder.filter_level(LevelFilter::Info);
     builder.init();
 
+    match connect() {
+        Ok(_) => (),
+        Err(_) => log::error!(target: "fp_extension", "Failed to connect to DB on init"),
+    };
+
     Extension::build().command("log", log).finish()
 }
 
-async fn connect() {
-    if MONGODB.get().is_some() {
-        log::warn!(target: "fp_extension", "Connection to DB already present!");
-        return;
-    }
-
-    log::info!(target: "fp_extension", "Connecting to DB!");
-
-    let _url = env::var("FP_EXTENSION_MONGO_DB_URL")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_URL").to_string());
-    let _db_name = env::var("FP_EXTENSION_MONGO_DB_DBNAME")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_DBNAME").to_string());
-
-    if let Ok(client_options) = ClientOptions::parse(_url).await {
-        // client_options.app_name = Some("FPArma Server Extension".to_string());
-        if let Ok(client) = Client::with_options(client_options) {
-            let _ = MONGODB.set(client.database(&_db_name[..]));
-            log::info!(target: "fp_extension", "Connected to DB!");
+fn connect() -> Result<(), ()> {
+    match MONGODB.get() {
+        // This should never happen
```

Comment:
I am not super happy with this, as it should probably panic. 
But this shouldn't be able to happen anyways.